### PR TITLE
Replace prompt with showPromptDialog

### LINKED ISSUE #5741: Push notification name still uses browser prompt()
## Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.

## The problem

When activating the "Push notification" setting, there is a prompt where the user can enter the name of the current device. This still uses the browser prompt:
Screenshot_2020 05 03_16h06m34s_010_

This is for example an issue, when using electron to wrap HA:
Screenshot_2020 05 03_16h09m11s_011_

Since there are custom HA dialogs available, the prompt should be replaced with that.

Also:
When no name is entered, the checkbox still becomes active. When the windows is reloaded the checkbox is inactive again. 

## Expected behavior

Use HA dialog instead of prompt()

## Steps to reproduce

1. Go to user profile
2. Activate "Push notification"
3. Prompt will appear

## Environment

- Home Assistant release with the issue: 0.109.x
- Last working Home Assistant release (if known): None
- Browser and browser version: Edge Chrome 83
- Operating system: Windows 10

## State of relevant entities

None

## Problem-relevant configuration

Relevant line of code:
https://github.com/home-assistant/frontend/blob/82f80db558ceaab1d1cbe54fbfe2e2165e7cacd5/src/components/ha-push-notifications-toggle.js#L91

## Javascript errors shown in your browser console/inspector

None

### PR DESCRIPTION
**Pull request by TheZoker**
## Breaking change
None

## Proposed change
Replace the outdated `prompt()` function for the notification device name dialog with a material dialog. 

Screenshot_2020 10 23_01h36m23s_010_

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #5741
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-push-notifications-toggle.js [hunk 1]
```diff
@@ -88,7 +91,11 @@ class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
         browserName = "chrome";
       }
 
-      const name = prompt("What should this device be called ?");
```
Comment (spacegaier) [inline src/components/ha-push-notifications-toggle.js:91]: Those texts should be localized / translatable.

File: src/components/ha-push-notifications-toggle.js [hunk 2]
```diff
@@ -88,7 +91,11 @@ class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
...
+      const name = await showPromptDialog(this, {
+        title: "What should this device be called ?",
+        inputLabel: "Please enter a name for the device",
```
Comment (spacegaier) [inline src/components/ha-push-notifications-toggle.js:96]: Those texts should be localized.

Comment (TheZoker) [reply src/components/ha-push-notifications-toggle.js:94]: What would be the appropriate path for the texts?

Comment (spacegaier) [reply src/components/ha-push-notifications-toggle.js:94]: There is already a group "[...].profile.push_notifications" which I think makes sense. So perhaps "[...].profile.push_notifications.add_device.title" and "[...].profile.push_notifications.add_device.text"

File: src/components/ha-push-notifications-toggle.js [hunk 3]
```diff
@@ -90,8 +90,8 @@ class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
       }
 
       const name = await showPromptDialog(this, {
-        title: "What should this device be called ?",
-        inputLabel: "Please enter a name for the device",
+        title: this.hass.localize("ui.panel.profile.push_notifications.add_device.title"),
+        inputLabel: this.hass.localize("ui.panel.profile.push_notifications.add_device.text"),
```
Comment (spacegaier) [inline src/components/ha-push-notifications-toggle.js:94]: You also need to add those to `src/translations/en.json` as part of this PR.

Comment (TheZoker) [reply src/components/ha-push-notifications-toggle.js:94]: Yes of course, totally forgot to push that file 🤦

**Review by spacegaier — APPROVED** [review]

File: src/translations/en.json [hunk 4]
```diff
@@ -2675,7 +2675,11 @@
           "error_load_platform": "Configure notify.html5.",
           "error_use_https": "Requires SSL enabled for frontend.",
           "push_notifications": "Push notifications",
-          "link_promo": "Learn more"
+          "link_promo": "Learn more",
+          "add_device": {
+            "title": "What should this device be called?",
+            "text": "Please enter a name for the device"
```
Comment (bramkragten) [inline src/translations/en.json:2681]:
```suggestion
            "text": "Device name"
```

File: src/translations/en.json [hunk 5]
```diff
@@ -2675,7 +2675,11 @@
           "error_load_platform": "Configure notify.html5.",
           "error_use_https": "Requires SSL enabled for frontend.",
           "push_notifications": "Push notifications",
-          "link_promo": "Learn more"
+          "link_promo": "Learn more",
+          "add_device": {
```
Comment (bramkragten) [inline src/translations/en.json:2679]:
```suggestion
          "add_device_prompt": {
```

File: src/components/ha-push-notifications-toggle.js [hunk 6]
```diff
@@ -88,7 +89,11 @@ class HaPushNotificationsToggle extends EventsMixin(PolymerElement) {
         browserName = "chrome";
       }
 
-      const name = prompt("What should this device be called ?");
+      const name = await showPromptDialog(this, {
+        title: this.hass.localize("ui.panel.profile.push_notifications.add_device.title"),
+        inputLabel: this.hass.localize("ui.panel.profile.push_notifications.add_device.text"),
+        inputType: "string"
```
Comment (bramkragten) [inline src/components/ha-push-notifications-toggle.js:95]:
```suggestion
```

**Review by bramkragten — APPROVED** [review]

**Pull request merged**